### DISCUSSION TITLE
Flexible column sizing

### DISCUSSION THREAD

**Discussion by dhythm** [TanStack/table#4683] [Ideas]
I'm started integrating from other DataGrid library to TanStack.
This library is nice, but I have a feature request.

`ColumnSizingColumnDef` accepts only `number` type of size/maxSize/minSize.
![image](https://user-images.githubusercontent.com/10891318/215012568-98aa1e78-eec1-4953-b6cd-47367f95ad9c.png)

It would be nicer if `string` is also accepted because we could pass much more style options like `auto`, `20%` and so on.
![image](https://user-images.githubusercontent.com/10891318/215013009-fb3fc9e9-abca-4b7d-8123-55e6ee4f176d.png)
![image](https://user-images.githubusercontent.com/10891318/215012384-8248f481-f7fa-4e74-89ff-e65979eb02cc.png)

**Comment by dhythm** [comment]
memo)
related #3192 (But the discussing point is different.)

**Reply by evbo** [reply]
I wonder, have you tried just styling the table yourself? This seems to be the cleanest approach nowadays anyway with tailwind: 

https://tailwindcss.com/docs/table-layout

some nice extras:
https://codepen.io/iKlsR/pen/pogNzNJ?editors=1000

or a step-by-step guide:
https://www.youtube.com/watch?v=S4MhQ6peq8A

a bit extreme... using divs with flexbox:
https://medium.com/@snowleo208/how-to-create-responsive-table-d1662cb62075


**Update**

I'm using tailwind template in the codepen above, but regardless in the css you simply set `table-fixed` and then in the headers `th` you set whatever width you want, or simply don't specify. Works great, with glitches only happen at extremely small screen sizes (so use overflow, etc.) . I don't mess with any tanstack width settings at all...

**Reply by dhythm** [reply]
Thanks for your comment.
I'm not using CSS stuff for the current work, so would like to resolve by passing props.
The previous DataGrid library can be achieved, so I want to do in DataGrid stuff.

But, thanks for telling me about the cleanest approach 👍
